lib/livebook/file_system/s3.ex

```diff
@@ -59,7 +61,8 @@ defmodule Livebook.FileSystem.S3 do
       :hub_id
     ])
     |> put_region_from_uri()
-    |> validate_required([:bucket_url, :region, :access_key_id, :secret_access_key, :hub_id])
+    |> validate_required([:bucket_url, :region, :hub_id])
+    |> Livebook.Utils.validate_mutual_inclusion([:access_key_id, :secret_access_key])
```

Comment:
If `:aws_credentials` return `:undefined`, then we need to require this parameter, no? But also, what happens when using Livebook Teams, and someone has `aws_credentials` configure but not someone else? 🤔

Reply:
Yes. I suppose it could check in with `aws_credentials` during validation. But since it makes an attempt to connect after validating the changeset anyway, the hub is going to find out if the configuration is good or not before it gets saved. At least when the UI is involved. I don't think the hub's `save_file_system/1` does any validation of its own.

Reply:
I see. Here we could provide a more specific validation though, saying those need to be filled as there are no credentials. Thoughts @jonatanklosko?

Reply:
The validation from the alternative branch looks good to me!